### DISCUSSION TITLE
Importing tw in all components

### DISCUSSION THREAD

**Discussion by eleretic** [ben-rogerson/twin.macro#182] [Q&A] [answered]
Hi!

I was wondering if there is a way/plan to not have to import `tw` in all components which we want to use the prop in?
With emotion, the `css` prop is injected through a babel plugin. https://github.com/emotion-js/emotion/tree/master/packages/babel-preset-css-prop

I think it would be pretty cool to have built in `tw` prop throughout all components. Are there recommendations against this?

Was that discussed before?

Thanks!

**Comment by ben-rogerson** [comment, marked as answer]
Heya William 👋 

I don't have any plans to avoid the 'twin.macro' import completely (like a vanilla babel plugin).
But you can currently use the `tw` (and `css`) prop by just importing the library like this:

```js
import 'twin.macro'

const InputWithTwProp = () => <input tw="border hover:border-black" />
```

If you ever use styled-components instead of emotion, you'll also need to make sure `"autoCssProp": true,` is in your twin config.

**Reply by eleretic** [reply]
Hi Ben,

Thanks for the suggestion! That said, I am looking for a way to not have to add yet another import in every file we use :) I found a sub-optimal solution with https://github.com/PavelDymkov/babel-plugin-auto-import#babel-plugin-auto-import 

Maybe I should try my hand at making a babel plugin

**Reply by ben-rogerson** [reply]
I recall someone trying something like this in the past but not getting too far with it.
They had some issues using a babel plugin to add imports for twin (another babel plugin). 
But you might get further with it, let me know how it goes 👍

**Comment by luknl** [comment]
Hey, they just released a npm [Babel plugin for twin](https://github.com/ben-rogerson/babel-plugin-twin)!
(more info [here](https://github.com/ben-rogerson/twin.macro/issues/247))
